connman/src/service.c

```diff
@@ -8430,6 +8449,8 @@ int __connman_service_ipconfig_indicate_state(struct connman_service *service,
 	case CONNMAN_SERVICE_STATE_IDLE:
 	case CONNMAN_SERVICE_STATE_FAILURE:
 		__connman_ipconfig_disable(ipconfig);
+		service->nameservers_ipv4_refcount = 0;
+		service->nameservers_ipv6_refcount = 0;
```

Comment:
just wondering: having seen that __connman_service_nameserver_del_routes() turns into nop on zero refcounts... is it ok to just clear these without further actions/checks?

Reply:
I was looking for a place to ensure the reset of these values since at this point the service is anyways completely down and should not have any routes set anymore so it would be safe to clear these just to be sure that they start from 0 when connecting the service. I'll need to check if this happens at any tests with some added debug.